src/Spectre.libGenetic.Tests/MutationOperatorTest.cpp

```diff
@@ -167,16 +167,31 @@ TEST_F(MutationTest, toggles_in_approximate_percentage_of_cases_for_specified_bi
     const auto expectedNumberOfToggles = NUMBER_OF_TRIALS * individual.size() * BIT_SWAP_RATE;
     const auto allowedMissCount = ALLOWED_MISS_RATE * expectedNumberOfToggles;
     unsigned numberOfToggles = 0;
-    for (unsigned i = 0; i < NUMBER_OF_TRIALS; ++i)
+    for (unsigned i = 0u; i < NUMBER_OF_TRIALS; ++i)
     {
         const auto last = Individual(std::vector<bool>(individual.begin(), individual.end()));
         individual = mutate(std::move(individual));
-        for (size_t j = 0; j < individual.size(); ++j)
+        for (size_t j = 0u; j < individual.size(); ++j)
         {
             numberOfToggles += last[j] != individual[j];
         }
     }
     EXPECT_LT(numberOfToggles, expectedNumberOfToggles + allowedMissCount);
     EXPECT_GT(numberOfToggles, expectedNumberOfToggles - allowedMissCount);
 }
+
+TEST_F(MutationTest, test_if_returns_original_if_fillup_is_too_low)
```

Comment:
This test is a little bit confusing. If `individual` is all true, please, name it so, since otherwise context is lost.

On the other hand, why do we need multiple trials here? If we set `BIT_SWAP_RATE` to `ALWAYS` (or `NEVER`, dependent on the content of `individual`), we can test it once - this behaviour is deterministic, right?

Reply:
done